src/plugins/lightscript.js

```diff
@@ -121,7 +121,16 @@ pp.parseArrayComprehension = function (node) {
   const loop = this.startNode();
   node.loop = this.parseForStatement(loop);
   this.expect(tt.bracketR);
-  return this.finishNode(node, "ArrayComprehension");
+  node.comprehends = "array";
+  return this.finishNode(node, "Comprehension");
```

Comment:
I'm OK either way (in fact, leaving it is less work for me on updating unit tests) but my thinking here is that in later ast transforms we may want a single node type. Especially since the code for both types of comprehensions is going to be essentially the same.